Info/handler.ts

```diff
@@ -1,27 +1,37 @@
 import * as express from "express";
 import { wrapRequestHandler } from "io-functions-commons/dist/src/utils/request_middleware";
 import {
+  IResponseErrorInternal,
   IResponseSuccessJson,
+  ResponseErrorInternal,
   ResponseSuccessJson
 } from "italia-ts-commons/lib/responses";
 import * as packageJson from "../package.json";
+import { checkApplicationHealth, HealthCheck } from "../utils/healthcheck";
 
 interface IInfo {
   version: string;
 }
 
-type InfoHandler = () => Promise<IResponseSuccessJson<IInfo>>;
+type InfoHandler = () => Promise<
+  IResponseSuccessJson<IInfo> | IResponseErrorInternal
+>;
 
-export function InfoHandler(): InfoHandler {
-  return async () => {
-    return ResponseSuccessJson({
-      version: packageJson.version
-    });
-  };
+export function InfoHandler(healthCheck: HealthCheck): InfoHandler {
+  return () =>
+    healthCheck
+      .fold<IResponseSuccessJson<IInfo> | IResponseErrorInternal>(
+        problems => ResponseErrorInternal(problems.join("\n\n")),
```

Comment:
Are we sure than any critical information come from `healthCheck` is filtered before sending it through the public API `Info`?

Reply:
Good point. For the case I tested I saw no problems (resource unavailable, wrong credentials). Still it's important to point out, thanks. 

Do you see any case which can be problematic? Please also consider that, for what concerns to functions, the endpoint is unauthenticated but not public, it's reachable only from our private network.

Reply:
We could anyway return a generic Error message of what type of error occurs `problems.map(_ => _.__source);` and logs the complete error for detail. What do you think?

Reply:
lgtm as soon as it'll be easy to troubleshoot. @gunzip thoughts?

Reply:
👍